File: frontend/src/app/core/models/recruitment.ts
export type JobStatus = 'draft' | 'open' | 'closed';
export type CandidateStatus = 'applied' | 'screening' | 'interview_scheduled' | 'offered' | 'hired' | 'rejected';

export interface Job {
  id: string;
  title: string;
  description: string;
  location: string | null;
  department_id: string | null;
  department?: { id: string; name: string } | null;
  status: JobStatus;
  opened_at: string | null;
  closed_at: string | null;
  created_at: string;
  candidate_count?: number;
  // alias for templates that use _count
  _count?: { candidates: number };
}

export interface Candidate {
  id: string;
  job_id: string;
  job?: { id: string; title: string };
  name: string;
  email: string;
  phone: string | null;
  resume_url: string | null;
  cover_letter: string | null;
  status: CandidateStatus;
  interview_at: string | null;
  notes: string | null;
  created_at: string;
}

File: backend/app/schemas/recruitment.py
from datetime import datetime
from pydantic import BaseModel, EmailStr
from app.schemas.common import BaseSchema


class DeptRef(BaseSchema):
    id: str
    name: str


class JobCreate(BaseModel):
    title: str
    description: str
    location: str | None = None
    department_id: str | None = None
    status: str = "draft"


class JobUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    location: str | None = None
    department_id: str | None = None
    status: str | None = None


class JobOut(BaseSchema):
    id: str
    title: str
    description: str
    location: str | None = None
    status: str
    opened_at: datetime | None = None
    closed_at: datetime | None = None
    department_id: str | None = None
    department: DeptRef | None = None
    candidate_count: int = 0
    created_at: datetime
    updated_at: datetime


class CandidateCreate(BaseModel):
    job_id: str
    name: str
    email: EmailStr
    phone: str | None = None
    resume_url: str | None = None
    cover_letter: str | None = None
    status: str = "applied"


class CandidateUpdate(BaseModel):
    status: str | None = None
    interview_at: datetime | None = None
    notes: str | None = None


class CandidateOut(BaseSchema):
    id: str
    name: str
    email: str
    phone: str | None = None
    resume_url: str | None = None
    cover_letter: str | None = None
    status: str
    interview_at: datetime | None = None
    notes: str | None = None
    job_id: str
    employee_id: str | None = None
    created_at: datetime
    updated_at: datetime
